### ISSUE TITLE
The `closest` filtertype on images isn't respected in viewers

### ISSUE THREAD

**Issue by expenses** [AcademySoftwareFoundation/MaterialX#1525]
Hi, I'm running into an issue where `image` nodes that are set to closest/nearest-neighbour texture filtering still use bilinear filtering in `MaterialXView`, `MaterialXGraphEditor` and `usdview` when USD is compiled with MaterialX support. 

![20230909_15h41m56s_grim](https://github.com/AcademySoftwareFoundation/MaterialX/assets/13566135/328fd871-aadd-4bb4-8d0c-21a6b3d9bdd8)

I'm not sure if this is the same issue for both MaterialX and USD that can be solved by a fix to MaterialX or if they're seperate.

**Comment by jstone-lucasfilm** [comment]
Good catch, @expenses, and this looks like an oversight that needs to be addressed in MaterialXRender and MaterialXRenderGlsl.

**Comment by mapetikea** [comment]
I think that the bug is also present in the OSL codegen. It doesn't seem to respect filters either?

**Comment by jstone-lucasfilm** [comment]
Thanks for this report, @expenses, and both the GLSL and OSL cases should now be addressed in the main branch.
